kubeflow/trainer/backends/localprocess/backend.py

```diff
@@ -0,0 +1,236 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import logging
+import string
+import tempfile
+import uuid
+import random
+from datetime import datetime
+from typing import List, Optional, Set, Union, Iterator
+
+from kubeflow.trainer.constants import constants
+from kubeflow.trainer.types import types
+from kubeflow.trainer.backends.base import ExecutionBackend
+from kubeflow.trainer.backends.localprocess.types import (
+    LocalProcessBackendConfig,
+    LocalBackendJobs,
+    LocalBackendStep,
+    LocalRuntimeTrainer,
+)
+from kubeflow.trainer.backends.localprocess.constants import local_runtimes
+from kubeflow.trainer.backends.localprocess.job import LocalJob
+from kubeflow.trainer.backends.localprocess import utils as local_utils
+
+logger = logging.getLogger(__name__)
+
+
+class LocalProcessBackend(ExecutionBackend):
+    def __init__(
+        self,
+        cfg: LocalProcessBackendConfig,
+    ):
+        # list of running subprocesses
+        self.__local_jobs: List[LocalBackendJobs] = []
+        self.cfg = cfg
+
+    def list_runtimes(self) -> List[types.Runtime]:
+        return local_runtimes
+
+    def get_runtime(self, name: str) -> types.Runtime:
+        runtime = next((rt for rt in local_runtimes if rt.name == name), None)
+        if not runtime:
+            raise ValueError(f"Runtime '{name}' not found.")
+
+        return runtime
+
+    def get_runtime_packages(self, runtime: types.Runtime):
+        if isinstance(runtime.trainer, LocalRuntimeTrainer):
+            return runtime.trainer.packages
+        else:
+            logger.debug("Trainer type isn't supported by LocalProcessBackend")
+            return []
+
+    def train(
+        self,
+        runtime: Optional[types.Runtime] = None,
+        initializer: Optional[types.Initializer] = None,
+        trainer: Optional[Union[types.CustomTrainer, types.BuiltinTrainer]] = None,
+    ) -> str:
+        # set train job name
+        train_job_name = random.choice(string.ascii_lowercase) + uuid.uuid4().hex[:11]
+        # localprocess backend only supports CustomTrainer
+        if not isinstance(trainer, types.CustomTrainer):
+            raise ValueError("CustomTrainer must be set with LocalProcessBackend")
+
+        # create temp dir
+        venv_dir = tempfile.mkdtemp(prefix=train_job_name)
+        logger.debug("operating in {}".format(venv_dir))
+
+        runtime.trainer = local_utils.get_runtime_trainer(
+            runtime_name=runtime.name,
+            venv_dir=venv_dir,
+            framework=runtime.trainer.framework,
+        )
+
+        # build training job command
+        training_command = local_utils.get_training_job_command(
+            trainer=trainer,
+            runtime=runtime,
+            train_job_name=train_job_name,
+            venv_dir=venv_dir,
+            cleanup=self.cfg.cleanup_venv,
```

Comment:
Can you keep name consistent please ?
```suggestion
            cleanup_venv=self.cfg.cleanup_venv,
```